ksql-engine/src/main/java/io/confluent/ksql/function/UdfTemplate.java

```diff
@@ -71,35 +80,54 @@ static String generateCode(final Method method, final String obj) {
       return null;
     }
 
+    if (clazz.isArray()) {
+      return fromArray(arg, clazz, index);
+    }
+
     // using boxed type is safe: long.class and Long.class are both of type Class<Long>
     // and this is a no-op for non-primitives
     final Class<? extends T> boxedType = Primitives.wrap(clazz);
     if (boxedType.isAssignableFrom(arg.getClass())) {
       return boxedType.cast(arg);
-    }
-
-    if (arg instanceof String) {
+    } else if (arg instanceof String) {
```

Comment:
Checkstyle doesn't compile without this change since it believes the cyclical complexity of the code is reduced when I add the `else` blocks. 

Even though my personal preference is to add the else blocks (I have to reason less about what the code inside the if statement does) I usually would defer to existing practices. In this case I think the reason is compelling (checkstyle fails otherwise) so perhaps we should encourage if/else instead of if/return.